pkg/controller/volume/attachdetach/reconciler/reconciler_test.go

```diff
@@ -1612,24 +1619,32 @@ func verifyVolumeReportedAsAttachedToNode(
 	nodeName k8stypes.NodeName,
 	isAttached bool,
 	asw cache.ActualStateOfWorld,
+	timeout time.Duration,
 ) {
-	result := false
-	volumes := asw.GetVolumesToReportAttached(logger)
-	for _, volume := range volumes[nodeName] {
-		if volume.Name == volumeName {
-			result = true
+	var result bool
+	var lastErr error
+	err := wait.PollUntilContextTimeout(context.TODO(), 50*time.Millisecond, timeout, true, func(context.Context) (done bool, err error) {
```

Comment:
setting `immediate` to `true` might mean that, we are not waiting for reconciler loop to turn? This could be problematic in case start state matches end state, since we will quit once we match the state.

Reply:
Fixed.